src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -327,15 +334,47 @@ internal PSVersionHashTable(IEqualityComparer equalityComparer) : base(equalityC
 
         /// <summary>
         /// Returns ordered collection with Keys of 'PSVersionHashTable'
+        /// We want see special order:
+        ///     1. PSVersionName
+        ///     2. PSEditionName
+        ///     3. Remaining properties in alphabetical order
         /// </summary>
         public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort(new PSVersionTableComparer());
```

Comment:
`PSVersionHashTable` can have a static readonly field to hold an instance of `PSVersionTableComparer`, so as to avoid creating a new comparer every time.